tests/http/test_matrixfederationclient.py

```diff
@@ -451,7 +451,7 @@ def test_client_does_not_retry_on_400_plus(self):
         self.failureResultOf(d)
 
     def test_client_sends_body(self):
-        defer.ensureDeferred(
+        defer.ensureDeferred(  # type: ignore[unused-awaitable]
```

Comment:
Should this have a `self.get_success` around it? Or is the `self.pump` below sufficient to ensure this gets called?

Reply:
`ensureDeferred` will run the async function up until the first blocking await, ie. it should be fine.